### PR TITLE
plugins/bufdelete: init

### LINKED ISSUE #1833: [PLUGIN REQUEST] bufdelete.nvim
| Field    | Description |
|----------|-------------|
| Plugin   | `[bufdelete.nvim](https://github.com/famiu/bufdelete.nvim)`  | 
| Homepage | `https://github.com/famiu/bufdelete.nvim`     | 
| Nixpkgs  | `[pkgs](https://search.nixos.org/packages?query=bufdelete)` | 

## Extra Information

I want this plugin for the user autocommands to open alpha when closing all buffers.
See: [https://github.com/goolord/alpha-nvim/discussions/85#discussioncomment-7994227](https://github.com/goolord/alpha-nvim/discussions/85#discussioncomment-7994227)

### PR DESCRIPTION
**Pull request by MattSturgeon**
Simple plugin with only one config option and a handful of commands & lua functions.

See its [README](https://github.com/famiu/bufdelete.nvim/?tab=readme-ov-file).

Fixes #1833

### CODE REVIEW

**Review by GaetanLepage — APPROVED** [review]
LGTM

File: plugins/utils/bufdelete.nix [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      Must be a function that takes no argument and returns a list of buffers.
+    '';
+  };
+
+  settingsExample = {
+    buf_filter = ''
+      function()
+        -- TODO: return a list of buffers
+        return { }
+      end
```
Comment (MattSturgeon) [inline plugins/utils/bufdelete.nix:67]: I can't find any real life examples: https://github.com/search?q=lang%3Alua+bufdelete_buf_filter&type=code

Anyone familiar with the lua buffers api haven any suggestions?

Comment (GaetanLepage) [reply plugins/utils/bufdelete.nix:67]: No sorry.
Not a big deal since apparently, even github cannot provide examples...

**Comment by MattSturgeon** [comment]
@mergifyio queue

**Pull request merged**